### PR TITLE
updating docs for topics section

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
Adds some narrative docs for getting started and a VM quickstart.

### What issues does this PR fix or reference?
Fixes: #94 

### Commits signed with GPG?
Yes

Please review [idem-azurerm's Contributing Guide](https://github.com/eitrtechnologies/idem-azurerm/blob/master/.github/CONTRIBUTING.md) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: docs/topics/gettingstarted.rst [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+Testing and Building Resources
+==============================
+Before you build the resources defined in the ".sls" file you may want to test what will happen when the state file is
+run. To do this, run idem with the ``--test`` option.
+
+.. code-block:: bash
+
+    (idemenv) $ idem state mytest.sls --test
+
+Once you determine that your state file with perform the intended operations, then you can build the defined resources
```
Comment (BellaNotte1986) [inline docs/topics/gettingstarted.rst:132]: with perform -> will perform

File: docs/topics/vmquickstart.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
+============================
+Quickstart - Virtual Machine
+============================
+This quickstart is guide is meant to give you a state file you can modify to deploy a simple Linux virtual machine
```
Comment (BellaNotte1986) [inline docs/topics/vmquickstart.rst:4]: is guide is -> guide is

File: docs/topics/vmquickstart.rst [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+Testing and Building Resources
+==============================
+Before you build the resources defined in the ".sls" file you may want to test what will happen when the state file is
+run. To do this, run idem with the ``--test`` option.
+
+.. code-block:: bash
+
+    (idemenv) $ idem state basicvm.sls --test
+
+Once you determine that your state file with perform the intended operations, then you can build the defined resources
```
Comment (BellaNotte1986) [inline docs/topics/vmquickstart.rst:77]: with perform-> will perform

File: docs/topics/vmquickstart.rst [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+this execution module command in order to get that information:
+
+.. code-block:: bash
+
+    (idemenv) $ idem exec azurerm.network.public_ip_address.get vmbasic001-pip0 rg-basicvm --output=nested
+
+Now you're ready to start building more resources in Azure! Consult the
+`Idem State Module Reference <../ref/states/all/index.html>`_ for YAML examples of the creation of specific resources.
+
+**Happy Idemizing!**
```
Comment (BellaNotte1986) [inline docs/topics/vmquickstart.rst:94]: Nice

Comment (nicholasmhughes) [reply docs/topics/vmquickstart.rst:94]: :smile:

**Pull request merged**